javascript/languages.js

```diff
@@ -0,0 +1,51 @@
+const rusLow = [
+  'ё', '1', '2', '3', '4', '5', '6', '7', '8', '9', '0', '-', '=', 'Backspace',
+  'Tab', 'й', 'ц', 'у', 'к', 'е', 'н', 'г', 'ш', 'щ', 'з', 'х', 'ъ', '\\', 'Del',
+  'Capslock', 'ф', 'ы', 'в', 'а', 'п', 'р', 'о', 'л', 'д', 'ж', 'э', 'Enter',
+  'Shift', 'я', 'ч', 'с', 'м', 'и', 'т', 'ь', 'б', 'ю', '.', 'Shift', '&uarr;',
+  'Ctrl', 'Win', 'Alt', '&mdash;', 'Alt', 'Ctrl', '&larr;', '&darr;', '&rarr;',
+];
+
+const rusUpper = [
+  'Ё', '1', '2', '3', '4', '5', '6', '7', '8', '9', '0', '-', '=', 'Backspace',
+  'Tab', 'Й', 'Ц', 'У', 'К', 'Е', 'Н', 'Г', 'Ш', 'Щ', 'З', 'Х', 'Ъ', '\\', 'Del',
+  'Capslock', 'Ф', 'Ы', 'В', 'А', 'П', 'Р', 'О', 'Л', 'Д', 'Ж', 'Э', 'Enter',
+  'Shift', 'Я', 'Ч', 'С', 'М', 'И', 'Т', 'Ь', 'Б', 'Ю', '.', 'Shift', '&uarr;',
+  'Ctrl', 'Win', 'Alt', '&mdash;', 'Alt', 'Ctrl', '&larr;', '&darr;', '&rarr;',
+];
+
+const rusShift = [
+  'Ё', '!', '"', '#', '$', '%', '^', '?', '*', '(', ')', '_', '+', 'Backspace',
+  'Tab', 'Й', 'Ц', 'У', 'К', 'Е', 'Н', 'Г', 'Ш', 'Щ', 'З', 'Х', 'Ъ', '|', 'Del',
+  'Capslock', 'Ф', 'Ы', 'В', 'А', 'П', 'Р', 'О', 'Л', 'Д', 'Ж', 'Э', 'Enter',
+  'Shift', 'Я', 'Ч', 'С', 'М', 'И', 'Т', 'Ь', 'Б', 'Ю', '?', 'Shift', '&uarr;',
+  'Ctrl', 'Win', 'Alt', '&mdash;', 'Alt', 'Ctrl', '&larr;', '&darr;', '&rarr;',
+];
+
+const enLow = [
+  '`', '1', '2', '3', '4', '5', '6', '7', '8', '9', '0', '-', '=', 'Backspace',
+  'Tab', 'q', 'w', 'e', 'r', 't', 'y', 'u', 'i', 'o', 'p', '[', ']', '\\', 'Del',
+  'Capslock', 'a', 's', 'd', 'f', 'g', 'h', 'j', 'k', 'l', ';', '\'', 'Enter',
+  'Shift', 'z', 'x', 'c', 'v', 'b', 'n', 'm', ',', '.', '/', 'Shift', '&uarr;',
+  'Ctrl', 'Win', 'Alt', '&mdash;', 'Alt', 'Ctrl', '&larr;', '&darr;', '&rarr;',
+];
+
+const enUpper = [
+  '`', '1', '2', '3', '4', '5', '6', '7', '8', '9', '0', '-', '=', 'Backspace',
+  'Tab', 'Q', 'W', 'E', 'R', 'T', 'Y', 'U', 'I', 'O', 'P', '[', ']', '\\', 'Del',
+  'Capslock', 'A', 'S', 'D', 'F', 'G', 'H', 'J', 'K', 'L', ';', '\'', 'Enter',
+  'Shift', 'Z', 'X', 'C', 'V', 'B', 'N', 'M', ',', '.', '/', 'Shift', '&uarr;',
+  'Ctrl', 'Win', 'Alt', '&mdash;', 'Alt', 'Ctrl', '&larr;', '&darr;', '&rarr;',
+];
+
+const enShift = [
+  '~', '!', '@', '#', '$', '%', '^', '?', '*', '(', ')', '_', '+', 'Backspace',
+  'Tab', 'Q', 'W', 'E', 'R', 'T', 'Y', 'U', 'I', 'O', 'P', '{', '}', '|', 'Del',
+  'Capslock', 'A', 'S', 'D', 'F', 'G', 'H', 'J', 'K', 'L', ';', '\'', 'Enter',
+  'Shift', 'Z', 'X', 'C', 'V', 'B', 'N', 'M', '<', '>', '?', 'Shift', '&uarr;',
+  'Ctrl', 'Win', 'Alt', '&mdash;', 'Alt', 'Ctrl', '&larr;', '&darr;', '&rarr;',
+];
+
+export {
+  rusLow, rusUpper, rusShift, enLow, enUpper, enShift,
+};
```

Comment:
nested objects can be used here:
```suggestion
const LANGUAGES = {
  RU: {
    LOW: [...],
    UPPER: [...],
    SHIFT: [...],
  },
  EN: {...}
}
```
it will help to get rid from if-else-if-else constructions in caps/shiftSwitch functions for example:
```suggestion
function switchKeys(lang, keys, shift, caps) {
  let mode = MODES.LOW;
  
  if (caps) {
    mode = MODES.UPPER;
  }
  if (shift) {
    mode = MODES.SHIFT;
  }

  const currentKeysCollection = LANGUAGES[lang][mode];
  ...
}
```

Reply:
I have fixed. Made code refactoring, changed the language switching algorithm. 
Created a switchKeys function (commit:  033bc268).